app/src/main/java/org/wikipedia/analytics/eventplatform/BreadCrumbCustomLogHelper.kt

```diff
@@ -0,0 +1,29 @@
+package org.wikipedia.analytics.eventplatform
+
+import android.view.View
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import org.wikipedia.R
+import org.wikipedia.WikipediaApp
+import org.wikipedia.feed.view.ListCardView
+
+object BreadCrumbCustomLogHelper {
+
+    fun getCustomNameForListItemView(view: View, position: Int): String {
+        var currentParent = view.parent
+        while (currentParent !is ListCardView<*>) {
+            if (currentParent.parent != null) {
+                currentParent = currentParent.parent
+            } else {
+                // ListItemView is not in a CardView
+                return BreadCrumbViewUtil.getReadableNameForView(view.parent as RecyclerView) + "." + position
+            }
+        }
+        return currentParent.javaClass.simpleName + "." + position
+    }
+
+    fun getCustomNameForLanguageListItem(view: View): String {
+        return WikipediaApp.instance.languageState.appLanguageCodes.toString() + "language_added:" +
+                view.findViewById<TextView>(R.id.language_subtitle).text.toString()
```

Comment:
Apologies, I meant to say that _any_ reference to "language list" within Breadcrumbs is too much, and should ideally be abstracted away.

In this instance, it looks like we're trying to get the text of the list item that corresponds to a language. Is there a way for us to generalize this for getting the text of _any_ list item, in any list?

Otherwise, if it's too difficult to get arbitrary text of any list item, then do we really need this data? i.e. do we actually need the names of the specific languages that were clicked?

Reply:
The click on the language item is the only way to add the language to app language list, and that needs to be measured... and i believe what the app languages list was before adding and the code of the language added are the only important things for the schema... 
Since not all recyclerview items might have a textview in them, I thought a more custom treatment can be applied here?
One other way we can do that is to directly send a message with this info when the event happens... as in OnClick() on the item..